src/Migrate/custom/New-AzMigrateServerReplication.ps1

```diff
@@ -464,6 +514,251 @@ public static int hashForArtifact(String artifact)
         if ($HasTargetAVZone) {
             $ProviderSpecificDetails.TargetAvailabilityZone = $TargetAvailabilityZone
         }
+
+        if ($HasSqlServerLicenseType)
+        {
+            $validLicenseSpellings = @{ 
+                NoLicenseType = "NoLicenseType";
+                PAYG = "PAYG";
+                AHUB = "AHUB"
+            }
+            $SqlServerLicenseType = $validLicenseSpellings[$SqlServerLicenseType]
+            $ProviderSpecificDetails.SqlServerLicenseType = $SqlServerLicenseType
+        }
+
+        if ($HasTag) {
+            $IllegalCharKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthValue = New-Object Collections.Generic.List[String]
+
+            if ($Tag.Count -gt 50)
+            {
+                throw "InvalidTags : Too many tags specified. Requested tag count - '$($VMTag.Count)'. Maximum number of tags allowed - '50'."
+            }
+
+            foreach ($x in $Tag.Keys)
+            {
+                 if ($x.length -gt 512)
+                 {
+                     $ExceededLengthKey.add($x)
+                 }
+
+                 if ($x -match "[<>%&\?/.]")
```

Comment:
Make the illegal char array a variable, and use it elsewhere. So if we change in one place, it should be good.

Reply:
But how can we distinguish the type of error.